acme_diags/plot/cartopy/lat_lon_vector_plot.py

```diff
@@ -0,0 +1,234 @@
+from __future__ import print_function
+
+import os
+import numpy as np
+import numpy.ma as ma
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import cartopy.crs as ccrs
+from cartopy.mpl.ticker import LongitudeFormatter, LatitudeFormatter
+from acme_diags.driver.utils.general import get_output_dir
+from acme_diags.plot import get_colormap
+
+plotTitle = {'fontsize': 11.5}
+plotSideTitle = {'fontsize': 9.5}
+
+# Position and sizes of subplot axes in page coordinates (0 to 1)
+panel = [(0.1691, 0.6810, 0.6465, 0.2258),
+         (0.1691, 0.3961, 0.6465, 0.2258),
+         (0.1691, 0.1112, 0.6465, 0.2258),
+         ]
+
+# Border padding relative to subplot axes for saving individual panels
+# (left, bottom, right, top) in page coordinates
+border = (-0.06, -0.03, 0.13, 0.03)
+
+
+def add_cyclic(var):
+    lon = var.getLongitude()
+    return var(longitude=(lon[0], lon[0] + 360.0, 'coe'))
+
+
+def get_ax_size(fig, ax):
+    bbox = ax.get_window_extent().transformed(fig.dpi_scale_trans.inverted())
+    width, height = bbox.width, bbox.height
+    width *= fig.dpi
+    height *= fig.dpi
+    return width, height
+
+
+def plot_panel(n, fig, proj, vars, clevels, cmap,
+               title, parameters, stats=None):
+
+    # vars[2] is the Euclidean Norm.
+    var = vars[2]
+    var = add_cyclic(var)
+    lon = var.getLongitude()
+    lat = var.getLatitude()
+    var = ma.squeeze(var.asma())
+    # vars[0] is Zonal.
+    var_x = add_cyclic(vars[0])#/var
+    # vars[1] is Meridional.
+    var_y = add_cyclic(vars[1])#/var
+    var_x = ma.squeeze(var_x.asma())
+    var_y = ma.squeeze(var_y.asma())
+    x  , y = np.meshgrid(lon,lat)
+    #skip=(slice(None,None,2),slice(None,None,2))
+   
+
+    # Contour levels
+    levels = None
+    norm = None
+    if len(clevels) > 0:
+        levels = [-1.0e8] + clevels + [1.0e8]
+        norm = colors.BoundaryNorm(boundaries=levels, ncolors=256)
+
+    # Contour plot
+    ax = fig.add_axes(panel[n], projection=proj)
+    ax.set_global()
+    cmap = get_colormap(cmap, parameters)
+    p1 = ax.contourf(lon, lat, var,
+                     transform=ccrs.PlateCarree(),
+                     norm=norm,
+                     levels=levels,
+                     cmap=cmap,
+                     extend='both',
+                     )
+    
+    ax.set_aspect('auto')
+    ax.coastlines(lw=0.3)
+
+    if n != 2:
+        # Normalized vector fields for test and ref, but not diff
+        # TODO: Scale the vectors properly.
```

Comment:
I tried changing the stride (e.g., the `5` in `x[::5, ::5]`), but that didn't look good either. I'm unclear on how to adjust the stride algorithmically based on different lat/lon. The plots linked on this review do not scale the vectors in any way.

Reply:
@chengzhuzhang if the pull request looks good (aside from the test data path), I can hand it back over to you to continue trying to scale the vectors properly.